x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneQueryEvaluator.java

```diff
@@ -387,17 +399,18 @@ public void close() {
     /**
      * Creates a vector where all positions correspond to elements that don't match the query
      */
-    protected abstract Vector createNoMatchVector(BlockFactory blockFactory, int size);
+    protected abstract Block createNoMatchBlock(BlockFactory blockFactory, int size);
 
     /**
      * Creates the corresponding vector builder to store the results of evaluating the query
      */
-    protected abstract T createVectorBuilder(BlockFactory blockFactory, int size);
+    protected abstract T createBlockBuilder(BlockFactory blockFactory, int size);
 
     /**
      * Appends a matching result to a builder created by @link createVectorBuilder}
      */
-    protected abstract void appendMatch(T builder, Scorable scorer) throws IOException;
+    protected abstract void appendMatch(T builder, Scorable scorer, int docId, LeafReaderContext leafReaderContext, Query query)
```

Comment:
technically speaking we are doing a bit more than just using Block instead of Vector in this PR.
we are also extending the list of `appendMatch` arguments and AFAICS no one is using `docId`, `leafReaderContext` and `query` yet. These I guess will be needed for the highlighter evaluator?

Reply:
Done in 1c9c756716387921d2fedc912f22f1eab7530d31